wrangler-storage/src/test/java/io/cdap/wrangler/store/recipe/RecipeStoreTest.java

```diff
@@ -0,0 +1,84 @@
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.wrangler.store.recipe;
+
+import com.google.common.collect.ImmutableList;
+import io.cdap.cdap.api.NamespaceSummary;
+import io.cdap.cdap.common.conf.Constants;
+import io.cdap.cdap.test.SystemAppTestBase;
+import io.cdap.cdap.test.TestConfiguration;
+import io.cdap.wrangler.dataset.recipe.RecipeAlreadyExistsException;
+import io.cdap.wrangler.dataset.recipe.RecipeRow;
+import io.cdap.wrangler.proto.recipe.v2.Recipe;
+import io.cdap.wrangler.proto.recipe.v2.RecipeId;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+public class RecipeStoreTest extends SystemAppTestBase {
+  @ClassRule
+  public static final TestConfiguration CONFIG = new TestConfiguration(Constants.Explore.EXPLORE_ENABLED, false);
+  private static RecipeStore store;
+
+  @BeforeClass
+  public static void setupTest() throws Exception {
+    getStructuredTableAdmin().createOrUpdate(RecipeStore.RECIPE_TABLE_SPEC);
+    store = new RecipeStore(getTransactionRunner());
+  }
+
+  @After
+  public void cleanupTest() {
+    store.clear();
+  }
+
+  @Test
+  public void testSaveNewRecipe() {
+    NamespaceSummary summary = new NamespaceSummary("n1", "", 10L);
+    RecipeId recipeId = new RecipeId(summary);
+    ImmutableList<String> directives = ImmutableList.of("dir1", "dir2");
+    Recipe recipe = Recipe.builder(recipeId.getRecipeId())
+      .setRecipeName("dummy-name")
+      .setDescription("dummy description")
+      .setCreatedTimeMillis(100L)
+      .setUpdatedTimeMillis(100L)
+      .setDirectives(directives)
+      .build();
+    RecipeRow recipeRow = RecipeRow.builder(recipe).build();
+    store.saveRecipe(recipeId, recipeRow);
+  }
+
+  @Test
+  public void testSaveRecipeWithDuplicateName() {
+    NamespaceSummary summary = new NamespaceSummary("n1", "", 10L);
+    RecipeId recipeId = new RecipeId(summary);
+    Recipe recipe = Recipe.builder(recipeId.getRecipeId()).setRecipeName("duplicate-name").build();
+    RecipeRow recipeRow = RecipeRow.builder(recipe).build();
+    store.saveRecipe(recipeId, recipeRow);
+
+    RecipeId newRecipeId = new RecipeId(summary);
+    Recipe newRecipe = Recipe.builder(newRecipeId.getRecipeId()).setRecipeName("duplicate-name").build();
+    RecipeRow newRecipeRow = RecipeRow.builder(newRecipe).build();
+    try {
+      store.saveRecipe(newRecipeId, newRecipeRow);
+      Assert.fail();
+    } catch (RecipeAlreadyExistsException e) {
+      // expected
```

Comment:
Good to assert true here and assert false after this catch block.

Reply:
Will resolve in next PR

Reply:
Based on the discussion we had, shall we change this to, 
@Test(expected = RecipeAlreadyExistsException.class) 
It will be more specific. This test case will go through even if there are other exceptions other than RecipeAlreadyExistsException

Reply:
Sure, will do in next PR